src/platform/plugins/shared/esql/server/extensions_registry/index.ts

```diff
@@ -105,14 +105,43 @@ export class ESQLExtensionsRegistry {
     return uniqBy(recommendedItems, uniqByProperty);
   }
 
-  private unsetRecommendedItems<T>(
+  private unsetRecommendedItems<T extends { name: string }>(
     map: Map<string, T[]>,
-    indexPattern: string,
-    activeSolutionId: SolutionId
+    items: T[],
+    activeSolutionId: SolutionId,
+    getIndexPattern: (item: T) => string | undefined,
```

Comment:
I dont think you need to pass this function. You can use the logic directly. I dont even think we use a private and public method either. Just have a public method. We dont need to complicate it.